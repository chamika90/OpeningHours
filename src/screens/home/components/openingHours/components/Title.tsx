import React from 'react';
import {StyleSheet, Text, View} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome5';
import {theme} from '../../../../../config/theme';
const {colors} = theme;

type TitleProps = {
  title: string;
};
const Title = ({title}: TitleProps) => {
  return (
    <View style={styles.header}>
      <View style={styles.titleContainer}>
        <View style={styles.headerIcon}>
          <Icon name="clock" size={18} color={colors.primaryIconColor} />
        </View>
        <Text testID="title" style={styles.title}>
          {title}
        </Text>
      </View>
      <View style={styles.line} />
    </View>
  );
};

export default Title;

const styles = StyleSheet.create({
  header: {
    alignItems: 'flex-start',
    marginBottom: 10,
    flexDirection: 'column',
  },
  titleContainer: {
    flexDirection: 'row',
  },
  title: {
    color: colors.primaryTextColor,
    fontSize: 24,
    fontFamily: 'Roboto-Bold',
  },
  headerIcon: {
    marginRight: 10,
    justifyContent: 'center',
  },
  line: {
    width: '100%',
    height: 1,
    marginTop: 10,
    backgroundColor: colors.primaryTextColor,
  },
});
