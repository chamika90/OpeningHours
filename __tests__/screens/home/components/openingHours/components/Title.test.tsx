import 'react-native';
import React from 'react';
import {render} from '@testing-library/react-native';
import Title from '../../../../../../src/screens/home/components/openingHours/components/Title';

describe('Title ', () => {
  it('should match the title snapshot', () => {
    const {toJSON} = render(<Title title="Open hours" />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render the title ', () => {
    const {getByTestId} = render(<Title title="Open hours" />);
    expect(getByTestId('title')).toBeTruthy();
  });
});
