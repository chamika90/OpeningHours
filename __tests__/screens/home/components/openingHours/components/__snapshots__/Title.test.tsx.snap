// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title  should match the title snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "flexDirection": "column",
      "marginBottom": 10,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "justifyContent": "center",
          "marginRight": 10,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#A1A2A4",
              "fontSize": 18,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome5Free-Regular",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {
              "fontWeight": "400",
            },
          ]
        }
      >
        ï€—
      </Text>
    </View>
    <Text
      style={
        Object {
          "color": "#202125",
          "fontFamily": "Roboto-Bold",
          "fontSize": 24,
        }
      }
      testID="title"
    >
      Open hours
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#202125",
        "height": 1,
        "marginTop": 10,
        "width": "100%",
      }
    }
  />
</View>
`;
